### PR TITLE
[WIP] Job announcement creation

### PR DESCRIPTION
**Pull request by mRoca**

### CODE REVIEW

File: src/SensioLabs/JobBoardBundle/Controller/JobController.php [hunk 1]
```diff
@@ -19,20 +24,114 @@ public function showAction()
     }
 
     /**
-     * @Route("/preview", name="job_preview")
+     * @Route("/post", name="job_post")
+     * @Template()
+     *
+     * @param Request $request
+     * @return array|\Symfony\Component\HttpFoundation\RedirectResponse
```
Comment (saro0h) [inline src/SensioLabs/JobBoardBundle/Controller/JobController.php]: Not necessary.

File: src/SensioLabs/JobBoardBundle/Controller/JobController.php [hunk 2]
```diff
@@ -19,20 +24,114 @@ public function showAction()
...
+                $entity = $this->get('session')->get('current_new_job');
+                $from_preview = true;
+            } else {
+                $this->get('session')->remove('current_new_job');
+            }
+        }
+
+        $form = $this->createCreateForm($entity);
+
+        if ($request->isMethod('POST')) {
```
Comment (saro0h) [inline src/SensioLabs/JobBoardBundle/Controller/JobController.php]: This if statement is redoundant with the ``$form->handleRequest($request);`` as it is already done in the form component.

**Comment by alOneh** [comment]
Can you keep this PR only for the first issue and split over commits not related to this first issue in separated PRs ?

File: src/SensioLabs/JobBoardBundle/Controller/JobController.php [hunk 3]
```diff
@@ -19,20 +24,105 @@ public function showAction()
...
+        ));
+
+        return $form;
+    }
+
+    /**
+     * @param Form $form
+     * @return array
+     */
+    private function getAllFormErrorMessages(Form $form)
```
Comment (alOneh) [inline src/SensioLabs/JobBoardBundle/Controller/JobController.php]: You don't need that, you can find a better way to handle form errors.

File: src/SensioLabs/JobBoardBundle/DataFixtures/ORM/LoadJobData.php [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+                ->setContract(Job::getContractTypesKeys()[rand(0, count(Job::getContractTypesKeys()) - 1)])
+                ->setDescription('<p><b>Lorem ipsum</b> dolor sit amet, consectetur adipiscing elit. In quis nulla augue. Phasellus enim eros, luctus a dapibus ut, auctor et urna. Donec eget egestas mi. Sed auctor semper elit, quis elementum orci pulvinar ut. Donec leo quam, elementum non vulputate ut, laoreet eu augue. Pellentesque tempus luctus porttitor. In id orci enim, quis gravida massa. Praesent consequat hendrerit consequat. Cras nulla elit, pulvinar a dapibus vitae, pellentesque...</p>')
+                ->setHowToApply('Send your CV to : <a href="mailto:contact@nocompany.com">contact@nocompany.com</a>');
+
+            $manager->persist($job);
+        }
+
+        $manager->flush();
+    }
+}
```
Comment (alOneh) [inline src/SensioLabs/JobBoardBundle/DataFixtures/ORM/LoadJobData.php]: Set up your IDE to "ensure line feed at file end on Save"

File: src/SensioLabs/JobBoardBundle/Entity/Job.php [hunk 5]
```diff
@@ -0,0 +1,425 @@
...
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\ORM\Mapping\Entity;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * Class Job
+ *
+ * @Entity
+ *
+ * @ORM\Table(name="jobs", indexes={
```
Comment (alOneh) [inline src/SensioLabs/JobBoardBundle/Entity/Job.php]: I think all fields are not supposed to have an index, what do you think ?

File: src/SensioLabs/JobBoardBundle/Entity/Job.php [hunk 6]
```diff
@@ -0,0 +1,425 @@
...
+    protected $updatedAt;
+
+
+    // ------------------------
+
+    /**
+     * @var array
+     */
+    public static $CONTRACT_TYPES = array(
+        ''           => "Type of contract",
```
Comment (alOneh) [inline src/SensioLabs/JobBoardBundle/Entity/Job.php]: We don't need this value here

File: src/SensioLabs/JobBoardBundle/Entity/Job.php [hunk 7]
```diff
@@ -0,0 +1,425 @@
...
+
+
+    // -----------------------------------------------
+
+
+    /**
+     * @param $text
+     * @return string
+     */
+    public static function slugify($text)
```
Comment (alOneh) [inline src/SensioLabs/JobBoardBundle/Entity/Job.php]: This method is not related to this entity

File: src/SensioLabs/JobBoardBundle/Resources/views/Includes/job_container.html.twig [hunk 8]
```diff
@@ -1,11 +1,11 @@
-{% for i in 0..5 %}
+{% for job in jobs %}
     {% embed 'SensioLabsJobBoardBundle:Includes:job.html.twig' %}
         {% block job_header %}
             {% include "SensioLabsJobBoardBundle:Includes:shortened_job_header.html.twig" %}
         {% endblock %}
 
         {% block job_description %}
-            <div class="padding description">Lorem ipsum dolor sit amet, consectetur adipiscing elit. In quis nulla augue. Phasellus enim eros, luctus a dapibus ut, auctor et urna. Donec eget egestas mi. Sed auctor semper elit, quis elementum orci pulvinar ut. Donec leo quam, elementum non vulputate ut, laoreet eu augue. Pellentesque tempus luctus porttitor. In id orci enim, quis gravida massa. Praesent consequat hendrerit consequat. Cras nulla elit, pulvinar a dapibus vitae, pellentesque...</div>
+            <div class="padding description">{{ job.description | striptags | slice(0, 420)  }}</div>
```
Comment (alOneh) [inline src/SensioLabs/JobBoardBundle/Resources/views/Includes/job_container.html.twig]: Twig provide you some nice shortcut to do that in extensions.

File: src/SensioLabs/JobBoardBundle/Entity/Job.php (hunk 5 above)
Comment (mRoca) [inline src/SensioLabs/JobBoardBundle/Entity/Job.php]: For the moment, we must permit to filter by country and contract type, but the project  allows people to find a job, so a full search form must be provided (later ?).

File: src/SensioLabs/JobBoardBundle/DataFixtures/ORM/LoadJobData.php (hunk 4 above)
Comment (mRoca) [inline src/SensioLabs/JobBoardBundle/DataFixtures/ORM/LoadJobData.php]: Done
